### PR TITLE
chore: update package json [DET-5846]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

- update package to remove any production audit security vulnerabilities
- update `dependencies` to the newest possible versions
- opportunistically update `devDependencies` (storybook in particular)

NOTE:
- `history` was left at `v4.10.1`, because `v5.x` has unresolvable breaking changes.
- `antd 4.x` introduced some breaking changes.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/package.json [hunk 1]
```diff
@@ -16,36 +16,34 @@
     "build-storybook": "NODE_PATH=src build-storybook -s public -o build-storybook"
   },
   "dependencies": {
-    "@types/react-window": "^1.8.3",
```
Comment (hkang1) [inline webui/react/package.json:19]: moved this one over to `devDependencies`

File: webui/react/src/components/ResponsiveTable.module.scss [hunk 2]
```diff
@@ -1,5 +1,5 @@
 .headerFilterOn {
-  :global(.ant-table-filter-column .ant-table-filter-trigger-container .ant-table-filter-trigger) {
+  :global(.ant-table-filter-column .ant-table-filter-trigger) {
```
Comment (hkang1) [inline webui/react/src/components/ResponsiveTable.module.scss:2]: the `ant-table-filter-trigger` structure has changed w/ antd 4.x

File: webui/react/src/pages/ExperimentDetails/ExperimentTrials.tsx [hunk 3]
```diff
@@ -300,7 +300,7 @@ const ExperimentTrials: React.FC<Props> = ({ experiment }: Props) => {
           rowSelection={{
             onChange: handleTableRowSelect,
             preserveSelectedRowKeys: true,
-            selectedRowKeys: settings.row,
+            selectedRowKeys: settings.row ?? [],
```
Comment (hkang1) [inline webui/react/src/pages/ExperimentDetails/ExperimentTrials.tsx:303]: although `selectedRowKeys` is supposed to take in an `Array` or `undefined`, when `undefined` is passed in with the newer `antd 4.13+` version, it crashes antd `Table`.

**Review by EmilyBonar — APPROVED** [review]
LGTM! Nice work!

**Comment by hkang1** [comment]
> LGTM! Nice work!

Thanks for the review as always!

**Pull request merged**